compiler/luci/service/CMakeLists.txt

```diff
@@ -21,5 +21,6 @@ nnas_find_package(GTest REQUIRED)
 
 GTest_AddTest(luci_service_test ${TESTS})
 target_include_directories(luci_service_test PRIVATE src)
+target_link_libraries(luci_service_test luci_pass)
```

Comment:
sequence of loading in CMakeList.txt is
```
add_subdirectory(service)
add_subdirectory(pass)
```
which when processing this file, `luci_pass` maybe not unavailable.

Reply:
I thought that `add_subdirectory` just adds directory to build and CMake will calculates the dependencies after all subdirectories are added... am I wrong..?

Reply:
Not sure... CI seems to work.. but I sometimes had trouble...

Reply:
You should remove this for safety. It's dependency problem.

Reply:
This is needed this because `ShapeInferenceRule.test.cpp` and `TypeInferenceRule.test.cpp` are using `luci/Pass`.
If I remove this, build fail.

FYI, if I used `target_link_libraries(luci_service PUBLIC luci_pass)`, configure step would be failed because cycle is created in the dependency order.

Reply:
> This is needed this because ShapeInferenceRule.test.cpp and TypeInferenceRule.test.cpp

We have to find another way

Reply:
Then the only solution is integrating these files to `luci/Pass/src/CircleShapeInferencePass.test.cpp` and `luci/Pass/src/CircleTypeInferencePass.test.cpp`.

Reply:
There are tests for each, maybe not all, Pass. That is scope of Pass.
Current existing codes want to test for each Rules(shape algorithms), but somehow began testing with the Pass,
which is strictly speaking, in another module that uses this module.

I think it would be better to do like
- Do Rule testing in as scope of Rule, do not use Pass
- We will do all testing in Pass and there will be no Rule testing

I think you may first disable (not delete) all the tests for doing your refactoring and enable tests one by one, someday...
But this may have risks of "is refactoring safe?"... well.. I think there are several other thests that may cover this...

Reply:
I think you are right and I thought about what how should I handle this.
And I found something interesting.

- All of tests in `CircleShapeInferenceRule.test.cpp` and `CircleTypeInferenceRule.test.cpp` are really simple cases. 
- Recipes in `res/TensorFlowLiteRecipes/` include both simple cases and more complicated cases.
- `luci_unit_readtest` and `luci_unit_writetest` are using those recipes for validating inference results.

So is it okay to proceed following steps?

- If simple cases in `CircleShapeInferenceRule.test.cpp` and `CircleTypeInferenceRule.test.cpp` are not included in `res/TensorFlowLiteRecipes`, create new recipes.
- Add those test cases in luci/tests/test.lst`
- Remove `CircleShapeInferenceRule.test.cpp` and `CircleTypeInferenceRule.test.cpp`